packages/datagateway-common/src/table/columnFilters/dateColumnFilter.component.tsx

```diff
@@ -79,74 +90,149 @@ const DateColumnFilter = (props: {
 
   return (
     <form>
-      <MuiPickersUtilsProvider utils={DateFnsUtils}>
-        <KeyboardDatePicker
-          clearable
-          allowKeyboardControl
-          style={{ whiteSpace: 'nowrap' }}
-          inputProps={{ 'aria-label': `${props.label} filter` }}
-          invalidDateMessage="Date format: yyyy-MM-dd."
-          KeyboardButtonProps={{
-            size: 'small',
-            'aria-label': `${props.label} filter from, date picker`,
-          }}
-          id={props.label + ' filter from'}
-          aria-hidden="true"
-          format="yyyy-MM-dd"
-          placeholder="From..."
-          value={startDate}
-          views={['year', 'month', 'date']}
-          maxDate={endDate || new Date('2100-01-01T00:00:00Z')}
-          maxDateMessage="Invalid date range"
-          color="secondary"
-          onChange={(date) => {
-            setStartDate(date);
-            updateFilter({
-              date,
-              prevDate: startDate,
-              otherDate: endDate,
-              startDateOrEndDateChanged: 'startDate',
-              onChange: props.onChange,
-            });
-          }}
-          okLabel={<span style={{ color: buttonColour }}>OK</span>}
-          cancelLabel={<span style={{ color: buttonColour }}>Cancel</span>}
-          clearLabel={<span style={{ color: buttonColour }}>Clear</span>}
-        />
-        <KeyboardDatePicker
-          clearable
-          allowKeyboardControl
-          style={{ whiteSpace: 'nowrap' }}
-          inputProps={{ 'aria-label': `${props.label} filter` }}
-          invalidDateMessage="Date format: yyyy-MM-dd."
-          KeyboardButtonProps={{
-            size: 'small',
-            'aria-label': `${props.label} filter to, date picker`,
-          }}
-          id={props.label + ' filter to'}
-          aria-hidden="true"
-          format="yyyy-MM-dd"
-          placeholder="To..."
-          value={endDate}
-          views={['year', 'month', 'date']}
-          minDate={startDate || new Date('1984-01-01T00:00:00Z')}
-          minDateMessage="Invalid date range"
-          color="secondary"
-          onChange={(date) => {
-            setEndDate(date);
-            updateFilter({
-              date,
-              prevDate: endDate,
-              otherDate: startDate,
-              startDateOrEndDateChanged: 'endDate',
-              onChange: props.onChange,
-            });
-          }}
-          okLabel={<span style={{ color: buttonColour }}>OK</span>}
-          cancelLabel={<span style={{ color: buttonColour }}>Cancel</span>}
-          clearLabel={<span style={{ color: buttonColour }}>Clear</span>}
-        />
-      </MuiPickersUtilsProvider>
+      {props.filterByTime ? (
+        <MuiPickersUtilsProvider utils={DateFnsUtils}>
+          <KeyboardDateTimePicker
+            clearable
+            allowKeyboardControl
+            style={{ whiteSpace: 'nowrap' }}
```

Comment:
It's probably worth extracting some of these props into a `commonProps` variable which we spread via `{...commonProps}` onto the 4 picker components we have to reduce code repetition